app/services/accounting/transaction_service.rb

```diff
@@ -43,7 +39,7 @@ def sync!
     def resource
       # where().first is used here to make sure we get the first Accounting::Transaction
       # according to resource ID every time
-      @resource ||= Accounting::Transaction.where(transaction_id: payload[:id]).first
+      @resource ||= Accounting::Transaction.find_or_initialize_by(transaction_id: payload[:id])
```

Comment:
Won't this reintroduce the void bug, where the wrong transaction sometimes gets returned?

Reply:
We should accept the transactions as long as the payment method exists in RL. If payment method doesn't exist, it will not accepted because we have this(https://github.com/CKDev/accounting/blob/master/app/services/accounting/transaction_service.rb#L101).

Reply:
@tmichaelreis I think you meant something related to void. So when RL create a void transaction, it will create a new void transaction record in Accounting and send void request to Authnet for original transaction. Does Authnet create a another new void transaction?

Reply:
No, authnet doesn't create a second void transaction, it voids the original. But there are two Accounting::Transactions in RedLine and only the original one has the information necessary to apply the void properly, so if the second one is selected here, the void doesn't get applied in RedLine.

Reply:
https://github.com/CKDev/accounting/pull/16

Reply:
@tmichaelreis updated the PR.